### PR TITLE
make some fors in to foreachs

### PR DESCRIPTION
**Pull request by John-Colvin**
`for fn in `git ls-files 'src/*.d'`; do sed -i -e's/for (size_t i = 0; i < \([[:alnum:].]\+\); i++)/foreach (const i; 0 .. \1)/' $fn; done`

with 2 reverts and then one other change that I will highlight.

### CODE REVIEW

**Comment by dkorpel** [comment]
You can't always do this transformation, you need to watch that `i` is not modified inside the loop.

**Comment by RazvanN7** [comment]
Yeah this is not going to fly. Each for needs to be handled carefully when translating to foreach because some for loops have a changing condition. For example `for(int i; i < members.length; i++)` ; if the members array deletes or adds new elements the foreach translation is going to result in segfaults. We've had such regressions in the past. I suggest closing this PR and try to judge on a case by case basis.

File: src/dmd/dmangle.d [hunk 1]
```diff
@@ -1030,7 +1030,6 @@ public:
             {
                 assert(i + 1 == args.dim); // must be last one
                 args = &va.objects;
-                i = -cast(size_t)1;
```
Comment (John-Colvin) [inline src/dmd/dmangle.d:1033]: this line was dead code anyway.

**Comment by RazvanN7** [comment]
Voila: `core.exception.AssertError@src/dmd/glue.d(614): Assertion failure`

**Comment by John-Colvin** [comment]
> Yeah this is not going to fly. Each for needs to be handled carefully when translating to foreach because some for loops have a changing condition. For example `for(int i=0; i < members.length; i++)` ; if the elements are added or deleted from the `members` array inside the loop body, the foreach translation is going to result in ICEs. We've had such regressions in the past. I suggest closing this PR and try to judge on a case by case basis.

fair point. I will split it in to smaller chunks so it's manageable to check each one manually.

**Pull request closed without merging**